### PR TITLE
Add remaining XCTS variables to Schwarzschild solution

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

These are the remaining variables that need to be evaluated to compute the XCTS equations. I factored them such that the analytic solution and analytic data classes can share about half of the implementations.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Comment by nilsvu** [comment]
@wthrowe and @moxcodes would you be willing to take this one?

File: src/PointwiseFunctions/AnalyticData/Xcts/CommonVariables.hpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        tmpl::size_t<3>, Frame::Inertial>,
+    Tags::ConformalRicciTensor<DataType, 3, Frame::Inertial>,
+    Tags::ConformalRicciScalar<DataType>,
+    ::Tags::deriv<gr::Tags::TraceExtrinsicCurvature<DataType>, tmpl::size_t<3>,
+                  Frame::Inertial>,
+    ::Tags::div<Tags::LongitudinalShiftBackgroundMinusDtConformalMetric<
+        DataType, 3, Frame::Inertial>>>;
+
+/*!
+ * \brief Implementations for variables that analytic-data classes can share
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticData/Xcts/CommonVariables.hpp:41]: Documentation not showing up.

**Review by wthrowe — CHANGES_REQUESTED** [review]
gcc is upset for some reason.

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/Xcts/Test_Schwarzschild.cpp [hunk 2]
```diff
@@ -80,46 +85,75 @@ void test_solution(const double mass,
...
+         "shift_background"},
+        {{{inner_radius, outer_radius}}}, std::make_tuple(mass), DataVector(5));
+    pypp::check_with_random_values<1>(
+        &SchwarzschildProxy::matter_source_variables, solution_proxy,
+        "Schwarzschild", {"energy_density", "stress_trace", "momentum_density"},
+        {{{inner_radius, outer_radius}}}, std::make_tuple(mass), DataVector(5));
+  }
+  {
+    INFO("Verify the solution solves the XCTS equations");
+    // Once the XCTS equations are implemented we will check here that the
```
Comment (moxcodes) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/Xcts/Test_Schwarzschild.cpp:178]: The 'common variables' from the second commit currently appear to be untested -- I'm fine with the test being rolled into an analytic solution that uses the common variables, but I think the computations performed by these inherited operators should then be tested.
Perhaps this is what you mean by this comment? I would have thought that at least a few of those calculations could be tested reasonably easily by comparing to pointwise GR calculations elsewhere, or with an expanded set of python tests, but perhaps I'm mistaken.

If that's an infeasibly large amount of work, I'll ask that you at least test here a few tags that don't evaluate to exactly `0.0`.

Comment (nilsvu) [reply tests/Unit/PointwiseFunctions/AnalyticSolutions/Xcts/Test_Schwarzschild.cpp:178]: I added a bunch of tests, though these quantities actually all evaluate to zero in isotropic Schwarzschild coordinates :D I'll add Painlevé-Gullstrand and isotropic Kerr-Schild coordinates in a follow-up PR where more quantities will be non-trivial. And once #2667 is merged I'll verify the solution actually numerically solves the XCTS equations, as the code comment says.

**Comment by nilsvu** [comment]
Posted a bunch of fixups. Thanks @wthrowe and @moxcodes!

**Comment by wthrowe** [comment]
Looks fine.

**Review by moxcodes — COMMENTED** [review]
looks good, go ahead and squash -- thanks for bolstering the test!

**Comment by nilsvu** [comment]
Rebased and squashed. Thanks for reviewing @wthrowe and @moxcodes!

**Review by moxcodes — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**